RFCs/0017-TestCase-FullyQualifiedName.md

```diff
@@ -1,49 +1,97 @@
+
+## Managed Tests that are Not Type/Method Based
+
+It is possible to create unit tests in managed code where tests are not necessarily based on types & methods. While this is somewhat uncommon, we need to be able to handle this situation gracefully. Some examples are gherkin-dotnet and custom discoverers in xUnit. In these examples, an externally parsed definition file may determine the tests. For these cases, we have defined the 'other' form of FQN. FQNs in this form will look similar to the following.
+
+    fqn://clr/o/{extension}/{other}/{url}/{segments}…
+
+The `'o'` indicates to the test window that we cannot interpret the contents of the FQN. The `{extension}` segment is to allow each adapter to add a custom string identifying the format. The rest of the URL is completely up to the interpretation of the test adapter. Type and method information will not be interpreted. Note that adapters using this format may lose hierarchy or code location benefits that are provided by the class/method FQN structure. There may be other ways to provide this information such as other properties on the test case. Those mechanisms are outside the scope of this document.
+
+## Legacy Compatibility
+
+We intend to implement a shared routine that can convert a legacy FQN into this standardized format. Throughout the TestWindow code, in all other locations, we would use the new standardized format. Any edge cases in interpretation can likely be addressed in the conversion routine. This does mean, however, that when handling test cases from adapters that do not support the format, we will need to allocate another property to save the original legacy FQN value. This is necessary so that when we call back into the adapter, we supply the original FQN rather than the new one. 
+
+Also, when working with unmanaged test adapters, we will need to handle the case of not having a standardized FQN. One way to support non-conforming adapters is to define an FQN format specifically for them so that we can still have them "standardized" without interpreting the contents.
+
+**Example**
+
+    fqn://compat/Class::Method::Test()
+
+## TestCase FQN Property Implementation
+
+We intend for test adapter implementors to use this new FQN format for the `FullyQualifiedName` property that is supplied to the TestCase object. Even though it is not specified, there may be some compatibility concerns with existing tools that depend on the `FullyQualifiedName` being in the current format. For an intermediate period, while compatibility concerns are worked through, we would suggest providing the new FQN in an additional custom property on the TestCase object named "StdFullyQualifiedName". Adding this additional property will affect the performance and allocation profile of the testing infrastructure, and we recommend not using this technique unless it is necessary for compatibility.
+
```

Comment:
The spec should probably also clarify whether a new field is needed for the new ID. I am guessing that TestCase discoverers (SBD, adapters) can just change the existing ID field to adhere to this spec. But I'm not sure if there are any compat concerns around that...

Reply:
Per recent discussion, we could probably make things work with a single field. If so, we should update this portion of the spec accordingly...

Reply:
I'll add the explanation regarding the runsetting.